actors/builtin/payment_channel/payment_channel_actor.go

```diff
@@ -0,0 +1,247 @@
+package payment_channel
+
+import (
+	"bytes"
+
+	addr "github.com/filecoin-project/go-address"
+	abi "github.com/filecoin-project/specs-actors/actors/abi"
+	big "github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	vmr "github.com/filecoin-project/specs-actors/actors/runtime"
+	indices "github.com/filecoin-project/specs-actors/actors/runtime/indices"
+	"github.com/filecoin-project/specs-actors/actors/serde"
+	"github.com/minio/blake2b-simd"
+)
+
+type PaymentChannelActor struct{}
+
+// type PaymentInfo struct {
+// 	PayChActor     address.Address
+// 	Payer          address.Address
+// 	ChannelMessage *cid.Cid
+
+// 	Vouchers []*types.SignedVoucher
+// }
+
+/////////////////
+// Constructor //
+/////////////////
+
+type PCAConstructorParams struct {
+	To addr.Address
+}
+
+func (pca *PaymentChannelActor) Constructor(rt vmr.Runtime, params *PCAConstructorParams) *vmr.EmptyReturn {
+
+	var st PaymentChannelActorState
+	rt.State().Transaction(&st, func() interface{} {
+		st.From = rt.ImmediateCaller()
+		st.To = params.To
+		st.LaneStates = make(map[int64]*LaneState)
+		return nil
+	})
+	return &vmr.EmptyReturn{}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Payment Channel state operations
+////////////////////////////////////////////////////////////////////////////////
+
+type PCAUpdateChannelStateParams struct {
+	Sv     SignedVoucher
+	Secret []byte
+	Proof  []byte
+}
+
+func hash(b []byte) []byte {
+	s := blake2b.Sum256(b)
+	return s[:]
+}
```

Comment:
Worth having a convo @anorth @whyrusleeping to decide whether to keep SHA256 or blake2 throughout actors. Best to stick to a single fn

Reply:
I suggest we make hashing a syscall rather than import a concrete implementation directly into actor code. @whyrusleeping ?